Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift

```diff
@@ -136,45 +136,12 @@ public final class JSONRPCConnection: Connection {
     state = .created
     self.messageRegistry = messageRegistry
 
-    let ioGroup = DispatchGroup()
-
-    #if os(Windows)
-    let rawInFD = dispatch_fd_t(bitPattern: inFD._handle)
-    #else
-    let rawInFD = inFD.fileDescriptor
-    #endif
-
-    ioGroup.enter()
-    receiveIO = DispatchIO(
-      type: .stream,
-      fileDescriptor: rawInFD,
-      queue: queue,
-      cleanupHandler: { (error: Int32) in
-        if error != 0 {
-          logger.fault("IO error \(error)")
-        }
-        ioGroup.leave()
-      }
-    )
+    self.ioGroup = DispatchGroup()
 
-    #if os(Windows)
-    let rawOutFD = dispatch_fd_t(bitPattern: outFD._handle)
-    #else
-    let rawOutFD = outFD.fileDescriptor
-    #endif
-
-    ioGroup.enter()
-    sendIO = DispatchIO(
-      type: .stream,
-      fileDescriptor: rawOutFD,
-      queue: sendQueue,
-      cleanupHandler: { (error: Int32) in
-        if error != 0 {
-          logger.fault("IO error \(error)")
-        }
-        ioGroup.leave()
-      }
-    )
+    self.inFD = inFD
+    self.outFD = outFD
+
+    self.ioGroup.enter()
 
     ioGroup.notify(queue: queue) { [weak self] in
```

Comment:
I think the entire `ioGroup` is now just a convoluted way of saying: This should be executed when the connection is closed: This will get executed when `ioGroup.leave()` is called, which happens after calling `closeAssumingOnQueue` in `JSONRPCConnection.close()` and the error handler of `JSONRPCConnectin.send`.

So, if we just executed this code inside `closeAssumingOnQueue`, I think we could get rid of the entire `ioGroup` logic and greatly simplify this.

Incidentally, now that I look at this, we might have been missing a few `ioGroup.leave()` before as well.